frontend/src/types/templates.ts

```diff
@@ -1,9 +1,26 @@
-interface Template {
+import {Organization, Page, User} from "./models.ts";
+
+export interface Template {
+    id: string;
     name: string;
-    lastUpdated: Date | undefined;
-    createdBy: string | undefined;
-    lab: string | undefined;
+    description: string;
+    labName: string
+    pages: Page[];
     status: TemplateStatus;
+    createdBy: User;
+    updatedBy: User | undefined;
+    createdAt: Date;
+    updatedAt: Date | undefined;
+    organization: Organization;
 }
 
-type TemplateStatus = "Completed"
\ No newline at end of file
+type TemplateStatus = "Completed" | "In Progress" | "Deprecated"
+
+const MIDDLEWARE_URL = import.meta.env.MIDDLEWARE_API_URL || 'http://localhost:8081';
+export const TemplateAPI = {
+    getTemplates: async (): Promise<Template[]> => {
+        const response = await fetch(`${MIDDLEWARE_URL}/templates`);
```

Comment:
should we wrap this in a try catch?
```
try{ 
--code above
}

catch(error){
console.error("Error in getting templates from API:",error);
return [];
}
```

Reply:
Actually, no. https://tanstack.com/query/v3/docs/framework/react/guides/query-functions#handling-and-throwing-errors This is one of the benefits of react query. 

We may want to add some code to check the error, but we don't want to wrap it here.

Reply:
ok sounds good!